AWSIoT/Internal/AWSIoTMQTTClient.m

```diff
@@ -639,6 +639,16 @@ - (void)openStreams:(id)sender
     [defaultRunLoopTimer invalidate];
     
     if (!self.runLoopShouldContinue ) {
```

Comment:
Good catch. The connectionAgeTimer will self terminate once the max age has been reached, but if a disconnect is issued before then, it will continue to run and hold on to the object reference. 

I think we can simplify this by just doing this after line 639. Thoughts?

         if (self.connectionAgeTimer != nil ) {
                [self.connectionAgeTimer invalidate];
                self.connectionAgeTimer = nil;
            }

Reply:
I believe we need to move the `invalidate` call because of the documentation below. 

From https://developer.apple.com/documentation/foundation/nstimer/1415405-invalidate?language=objc 

> Special Considerations
> You must send this message from the thread on which the timer was installed. If you send this message from another thread, the input source associated with the timer may not be removed from its run loop, which could prevent the thread from exiting properly.